### PR TITLE
Map

### PR DESCRIPTION
**Pull request by AC159**
### Part 1: Map implementation

**This part can now:**

- Load a map file and validate if the syntax is correct
- Create a map object out of this file
- Validate that the map data structure is a connected graph

### CODE REVIEW

File: WarzoneMaps/solar/invalidsmallsolar.map [hunk 1]
```diff
@@ -0,0 +1,28 @@
+; map: canada.map
```
Comment (kikkiii21) [inline WarzoneMaps/solar/invalidsmallsolar.map:1]: we will have to either remove that (its referencing another map) or put something else. In any case, it shouldnt affect the reading of the map

File: WarzoneMaps/solar/smallsolar.map [hunk 2]
```diff
@@ -0,0 +1,28 @@
+; map: canada.map
```
Comment (kikkiii21) [inline WarzoneMaps/solar/smallsolar.map:1]: same thing here. It's a minor fix

File: WarzoneMaps/solar/smallsolarduplicates [hunk 3]
```diff
@@ -0,0 +1,28 @@
+; map: canada.map
```
Comment (kikkiii21) [inline WarzoneMaps/solar/smallsolarduplicates:1]: same minor fix

File: WarzoneMaps/solar/smallsolarduplicates.map [hunk 4]
```diff
@@ -0,0 +1,28 @@
+; map: canada.map
```
Comment (kikkiii21) [inline WarzoneMaps/solar/smallsolarduplicates.map:1]: same minor fix

**Review by kikkiii21 — APPROVED** [review]
looks great! Can't wait to start integrating all the parts together

File: WarzoneMaps/solar/invalidsmallsolar.map (hunk 1 above)
Comment (AC159) [reply WarzoneMaps/solar/invalidsmallsolar.map:1]: We can leave it since it is not read by anything in the code

**Pull request merged**